src/gordon_gcp/clients/http.py

```diff
@@ -168,3 +182,27 @@ def __init__(self, auth_client=None, session=None):
             json_callback = json.loads
         response = await self.request(method='get', url=url, **kwargs)
         return json_callback(response)
+
+    async def get_all(self, url, params):
+        """Aggregate data from all pages of a API query.
+
+        Args:
+            url (str): Google API endpoint URL.
+            params (dict): URL query parameters.
+
+        Returns:
+            list: Parsed JSON query response results.
+        """
+        items = []
+        next_page_token = None
+
+        while True:
+            if next_page_token:
+                params['pageToken'] = next_page_token
+            response = await self.get_json(url, params=params)
+
+            items.append(response)
```

Comment:
Is append here?  I'm not sure about the format, but maybe extend?  Or even something further down the tree?